src/preprocessing/passes/bool_to_bv.cpp

```diff
@@ -221,7 +215,157 @@ void BoolToBV::lowerNode(const TNode& n)
     default: break;
   }
 
+  // check if it's safe to lower or rebuild the node
+  // Note: might have to rebuild to keep changes to children, even if this node
+  // isn't being lowered
+
+  // it's safe to lower if all the children are bit-vectors
+  bool safe_to_lower =
+      (new_kind != k);  // don't need to lower at all if kind hasn't changed
+
+  // it's safe to rebuild if rebuilding doesn't change any of the types of the
+  // children
+  bool safe_to_rebuild = true;
+
+  for (const Node& nn : n)
+  {
+    if (safe_to_lower)
```

Comment:
(optional) You could do `safe_to_lower = safe_to_lower && ...;`

My preference is generally to minimize explicit control flow, which is why I suggest this.

Reply:
Good point, done with 516680e786d4ac6bf38d1771cdbd1f250028203a.